projects/FHC-AtmelStudio/Fussl-01/Fussl-01/f_range_int.h

```diff
@@ -0,0 +1,151 @@
+/*
+* f_range_int.h
+*
+* Created: 13.03.2018 17:52:31
+* @author Maximilian Starke
+*/
+
+#ifndef __F_RANGE_INT_H__
+#define __F_RANGE_INT_H__
+namespace fsl {
+	namespace ver_1_0 {
+		namespace lg {
+			
+			/*! @brief Type for Integers in a range {\a base_type_constants::ZERO, ... , \a base_type_constants::MAX}
+			@details Stores a number in {\a base_type_constants::ZERO, ... , \a base_type_constants::MAX} or #OUT_OF_RANGE.
+			@tparam _base_type The base integer type that is used internally. !add description, what this type must bring with it. Must provide a total order with comparison operator < and operator ==. ??It must provide constructors for _base_type(0), _base_type(1), _base_type(-1)
+			@tparam _RANGE The maximum value that a #fsl::ver_1_0::lg::range_int can take + 1. Must be positive.
+			@tparam _OVERFLOW_RESULTS_IN_OUT_OF_RANGE Defines whether some arithmetic operation (e.g. #operator+) leading to an overflow will result in #OUT_OF_RANGE or if it will behave like a ring in mathematics.
+			@tparam _OUT_OF_RANGE_IS_ABSORBING If it is set true any arithmetical operation will result in #OUT_OF_RANGE if one of the operands is #OUT_OF_RANGE.
+			*/
+			template <typename _base_type, _base_type _RANGE, bool _OVERFLOW_RESULTS_IN_OUT_OF_RANGE = true, bool _OUT_OF_RANGE_IS_ABSORBING = true>
+			class range_int {
+				public:
+				
+				/// @brief \range_int__base_type
+				using base_type = _base_type;
+				
+				/// @brief Contains all constants of type #base_type.
+				struct base_type_constants {
+					
+					/// @brief Zero in base_type.
+					static constexpr base_type ZERO{ base_type(0) };
+					
+					/// @brief One in base_type.
+					static constexpr base_type ONE{ base_type(1) };
+					
+					/// \range_int__RANGE
+					static constexpr base_type RANGE{ _RANGE };
+					static_assert(ZERO < RANGE, "Invalid template argument: Range limit must be a positive value.");
+					
+					/// @brief Minimum value that a #fsl::ver_1_0::lg::range_int can take.
+					static constexpr base_type MIN{ ZERO };
+					
+					/// @brief Maximum value that a #fsl::ver_1_0::lg::range_int can take.
+					static constexpr base_type MAX{ RANGE - 1 };
+					
+					/// @brief A dedicated value of type #base_type that is interpreted as out of range.
+					static constexpr base_type OUT_OF_RANGE{ base_type(-1) };
+					static_assert(!(ZERO < OUT_OF_RANGE && OUT_OF_RANGE < RANGE), "Internal error: The base_type value for OUT_OF_RANGE must not be part of the range.");
+				};
+				
+				/// \range_int__RANGE
+				static constexpr base_type RANGE{ base_type_constants::RANGE };
```

Comment:
Because of simplification (reducing number of template arguments by removing some [What is the behavior in overflow case]- flags) only this one member remains. So I'll keep it as it is.